legit.py

```diff
@@ -0,0 +1,230 @@
+        a .git directory with subdirectories for objects,
+        refs/heads, refs/tags, and template files.
+        An initial HEAD file that references the HEAD
+        of the master branch is also created.
+
+        Note:
+            Git accordance is "git init `path`"
+            https://git-scm.com/docs/git-init
+
+        Args:
+            path: The path to initialize repo as git repository.
+
+        Returns:
+            The return value is path to repo. Str for success, None otherwise.
+
+        Examples:
+            >>> path_to_repo = Repository.init(path).repo
+
+        """
+        log.debug(f"git init {path}")
+        try:
+            Repo.init(path)
+        except GitCommandError:
+            logging.exception(f"GitCommandError: git init {GitCommandError} traceback")
+            return None
+        return cls(path)
+
+    def checkout(self, branch_name: str) -> None:
+        """Updates files in the working tree to match
+        the version in the index or the specified tree.
+        If no paths are given, git checkout will also update HEAD
+         to set the specified branch as the current branch.
+
+        Note:
+            Git accordance is "git checkout -b `branch_name`"
+            https://git-scm.com/docs/git-checkout
+
+        Args:
+            branch_name: The name for branch you want to create/switch.
+
+        Examples:
+            >>> path_to_repo = Repository.init(path).repo
+            >>> path_to_repo.checkout(`branch_name`)
+
+        """
+        log.debug(f"git checkout -b {branch_name}")
+        try:
+            self.repo.git.checkout('-b', branch_name)
+        except GitCommandError:
+            logging.exception(f"GitCommandError: git checkout {branch_name} traceback")
+            self.repo.git.checkout(branch_name)
+
+    def commit(self, message: str) -> None:
+        """Create a new commit containing the current contents
+        of the index and the given log message describing the changes.
+
+        Note:
+            Git accordance is "git commit -m `message`"
+            https://git-scm.com/docs/git-commit
+
+        Args:
+            branch_name: The name for branch you want to create/switch.
+
+        Examples:
+            >>> path_to_repo.commit(`message`)
+
+        """
+        log.debug(f"git commit -m \"{message}\"")
+        try:
+            self.repo.index.commit(message)
+        except GitCommandError:
+            logging.exception(f"GitCommandError: git commit {GitCommandError} traceback")
+
+    def add_changes(self, files: Iterable[str] = None) -> None:
+        """This command updates the index using
+        the current content found in the working tree,
+         to prepare the content staged for the next commit.
+
+        Note:
+            Git accordance is "git add `files`" or "git add ."
+            https://git-scm.com/docs/git-add
+
+        Args:
+            files: The path to files to add or None to add all
```

Comment:
judging by the code below, this is not a path but rather a sequence of file names? 